model/repo_ref.go

```diff
@@ -74,49 +50,18 @@ func (r *RepoRef) Insert() error {
 }
 
 // Upsert updates the project ref in the db if an entry already exists,
-// overwriting the existing ref. If no project ref exists, one is created
+// overwriting the existing ref. If no project ref exists, one is created.
+// Ensures that fields that aren't relevant to repos aren't set.
 func (r *RepoRef) Upsert() error {
```

Comment:
This is a slightly unrelated change but, I realized how prone this is to bugs. The only reason I handled all of the sets individually is because I didn't want the fields that didn't relate to repos to be set; however there are _much_ fewer of them so it makes sense to handle them separately instead.